packages/backend-deployer/src/cdk_deployer.ts

```diff
@@ -128,16 +198,50 @@ export class CDKDeployer implements BackendDeployer {
     backendId: BackendIdentifier,
     destroyProps?: DestroyProps
   ) => {
-    const cdkCommandArgs: string[] = ['--force'];
-    if (destroyProps?.profile) {
-      cdkCommandArgs.push('--profile', destroyProps.profile);
+    // eslint-disable-next-line no-console
+    console.log(destroyProps?.profile);
+
+    let assembly;
+    try {
+      assembly = await this.cdkToolkit.fromAssemblyDirectory(
+        this.relativeCloudAssemblyLocation
+      );
+      // eslint-disable-next-line amplify-backend-rules/no-empty-catch
+    } catch {
+      // do nothing
     }
-    return this.tryInvokeCdk(
-      InvokableCommand.DESTROY,
-      backendId,
-      this.getAppCommand(),
-      cdkCommandArgs
-    );
+
+    if (!assembly) {
+      // Looks like the sandbox has not been deployed yet. Let's try to create a new one
+      const contextParams: {
+        [key: string]: unknown;
+      } = {};
+      contextParams[CDKContextKey.BACKEND_NAMESPACE] = backendId.namespace;
+      contextParams[CDKContextKey.BACKEND_NAME] = backendId.name;
+      contextParams[CDKContextKey.DEPLOYMENT_TYPE] = backendId.type;
+      assembly = await this.cdkToolkit.fromAssemblyBuilder(
+        async () => {
+          await tsImport(
+            pathToFileURL(this.backendLocator.locate()).toString(),
+            import.meta.url
+          );
+          return new CloudAssembly(this.relativeCloudAssemblyLocation);
```

Comment:
afaik currently this does run synth (with ipc implementation). should we synth here?

Reply:
we don't have access to the `app` here and `synth` only exists on the `app` interface.

Reply:
I mean `amplifySynth` via events.

To test if we need this we can:
1. `npx ampx sandbox --once`
2. Nuke `.amplify` folder (or have fresh copy of the project by any other mean)
3. `npx ampx sandbox delete` <-- see if this works or not.

Reply:
ohh I misunderstood, I thought this comment was in `deploy` method. Yeah I'll test and add if synth is needed here.